src/Microsoft.DotNet.ProjectModel/ProjectReader.cs

```diff
@@ -132,6 +132,37 @@ public Project ReadProject(Stream stream, string projectName, string projectPath
                     throw new FormatException("The assembly file version is invalid: " + fileVersion, ex);
                 }
             }
+            
+            var analyzerOptionsJson = rawProject.Value("analyzerOptions") as JsonObject;
+            if (analyzerOptionsJson != null)
+            {
+                var analyzerOptions = new AnalyzerOptions();
+                
+                foreach (var key in analyzerOptionsJson.Keys)
+                {
+                    switch (key)
+                    {
+                        case "languageId":
```

Comment:
I'm not sure. For example, "csc" could mean both the Roslyn csc or the native csc but the native compiler doesn't support analyzers. More importantly, someone could create a Roslyn-compatible C# compiler that is not named csc. I think the `languageId` is separate from the compiler name.